components/raftstore/src/store/fsm/apply.rs

```diff
@@ -771,8 +762,8 @@ where
     /// are all handled.
     pending_msgs: Vec<Msg<EK>>,
 
-    /// Approximate size of current buffer.
-    approximate_size: usize,
+    /// Cache heap size for itself.
+    heap_size: Option<usize>,
```

Comment:
Why do we need an option here?

Reply:
Indicates whether it has been already calculated or not. It's possible that it's `Some(0)`, for which case we don't need to do the calculation again.